src/care/DebugPlugin.cpp

```diff
@@ -0,0 +1,139 @@
+#include "care/DebugPlugin.h"
+#include "chai/ExecutionSpaces.hpp"
+#include "care/CHAICallback.h"
+#include "care/PluginData.h"
+
+// Std library headers
+#if defined(CARE_DEBUG) && !defined(_WIN32)
+#include <execinfo.h>
+#endif // defined(CARE_DEBUG) && !defined(_WIN32)
+
+#include <unordered_set>
+
+namespace care{
+   void DebugPlugin::registerPlugin() {
+      static RAJA::util::PluginRegistry::add<care::DebugPlugin> L ("Debug plugin", "CARE plugin for debugging");
+   }
+
+   void DebugPlugin::preLaunch(const RAJA::util::PluginContext& p) {
+#if !defined(CHAI_DISABLE_RM)
+      // Prepare to record CHAI data
+      if (CHAICallback::isActive()) {
+         PluginData::clearActivePointers();
+
+#if defined(CARE_GPUCC) && defined(CARE_DEBUG)
+         GPUWatchpoint::setOrCheckWatchpoint<int>();
+#endif // defined(CARE_GPUCC) && defined(CARE_DEBUG)
+      }
+#endif // !defined(CHAI_DISABLE_RM)
+   }
+
+
+   void DebugPlugin::postLaunch(const RAJA::util::PluginContext& p) {
+#if !defined(CHAI_DISABLE_RM)
+	   chai::ExecutionSpace space;
+
+	   switch (p.platform) {
+         case RAJA::Platform::host:
+         	space = chai::CPU; break;
```

Comment:
The indentation is strange in this section.  Are there tabs or is this the case of a beautifier not understanding #ifdefs?

Reply:
I think this still needs to be fixed. And I'll give myself a TODO to figure out how to add a beautifier to standalone CARE.

Reply:
@neelakausik, can you fix the indentation?

Reply:
Thanks!